components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSOAPUtils.java

```diff
@@ -0,0 +1,281 @@
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     *
+     * @param samlRes
+     * @param acUrl
+     * @return
+     */
+    public static String createSOAPMessage(String samlRes, String acUrl) {
+        SOAPMessage soapMsg = null;
+        try {
+            MessageFactory factory = MessageFactory.newInstance();
+            soapMsg = factory.createMessage();
+            SOAPPart part = soapMsg.getSOAPPart();
+            SOAPEnvelope envelope = part.getEnvelope();
+            SOAPHeader header = envelope.getHeader();
+            SOAPHeaderElement soapHeaderElement = header.addHeaderElement(envelope.createName("Response",
+                    "ecp",
+                    "urn:oasis:names:tc:SAML:2.0:profiles:SSO:ecp"));
+            soapHeaderElement.setMustUnderstand(true);
+            soapHeaderElement.setActor("http://schemas.xmlsoap.org/soap/actor/next");
+            soapHeaderElement.addAttribute(new QName("AssertionConsumerServiceURL"),
+                    acUrl);
+            SOAPBody body = envelope.getBody();
+            String rawxml = "<![CDATA[" + samlRes + "]]>";
+            body.addTextNode(rawxml);
+        } catch (SOAPException e) {
+            log.error("SOAP Exception when creating SOAP Response");
+        }
+        if (soapMsg != null) {
+            return convertSOAPToString(soapMsg).replace("<![CDATA[", "").replace("]]>", "");
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     *Converts a  SOAP Message to String.
+     * @param soapMessage
+     * @return
+     */
+    public static String convertSOAPToString(SOAPMessage soapMessage) {
+        final StringWriter stringWriter = new StringWriter();
+        try {
+            TransformerFactory.newInstance().newTransformer().transform(
+                    new DOMSource(soapMessage.getSOAPPart()),
+                    new StreamResult(stringWriter));
+        } catch (TransformerException e) {
+            throw new RuntimeException(e);
+        }
+        return stringWriter.toString();
+    }
+
+    /**
+     *Send the SOAP fault with the servlet response.
+     * @param resp Servlet response
+     * @param faultsring SOAP Fault code
+     * @param faultcode SOAP fault code
+     */
+    public static void sendSOAPFault(HttpServletResponse resp, String faultsring , String faultcode) {
+        PrintWriter out = null;
+        try {
+            out = resp.getWriter();
+        } catch (IOException e) {
+            log.error("An IO Exception Occured ");
+        }
+        String soapFault = SAMLSOAPUtils.createSOAPFault(faultsring, faultcode);
+        log.error(soapFault);
+        resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        resp.setContentType("text/html;charset=UTF-8");
+        out.print(soapFault);
+    }
+
+    public static void doBootstrap() {
+        if (!isBootStrapped) {
```

Comment:
Aren't we already bootstrapping when the bundle is getting initiated ?

Reply:
> Aren't we already bootstrapping when the bundle is getting initiated ?

removed